include/usb/class/usb_hid.h

```diff
@@ -430,6 +404,35 @@ enum hid_kbd_led {
 	HID_KBD_LED_KANA	= 0x10,
 };
 
+#if defined(CONFIG_USB_DEVICE_HID)
```

Comment:
If doing it this way then this define should be defined in zephyr.doxyfile.in so that it is part of the documentation build.
Why is this better than the first approach to have it in a separate header file?

Reply:
I would have preferred a common header that could be included by both, and avoid the use of `#if`. It just complicates the header, and the documentation build.

Reply:
Disagree, if Bluetooth project uses defines from USB HID class then it has to include the header from `include/usb/class/`. Would you be more comfortable to split `include/usb/usb_hid.h` in` include/usb/class/usb_hid.h` and `include/usb/class/usb_hid_device.h`?

Reply:
I will second Carles' suggestion. I think HID over USB or GATT should be treated equally.
But splitting it as you suggest would be acceptable as well, it's better than having '#if' in the header.